### ISSUE TITLE
[bug]: Save option overrides the request name

### ISSUE THREAD

**Issue by kyteinsky** [hoppscotch/hoppscotch#2132]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What I've been going through:

When I have a REST request saved in a collection with a certain name `xyz`, and I change some parameters of the request and then save it again the name of the request inside the collection is changed to `Untitled request`.

### What I expected to happen:

The request should've been saved with the same name (`xyz`).

### Steps to reproduce

1. Save a REST request with a name other than `Untitled request`
2. Change the request
3. Save it again

### Environment

Production

### Version

Local

**Comment by AndrewBastin** [comment]
Huh, I wasn't able to reproduce this on my end.

@liyasthomas  ?

**Comment by kyteinsky** [comment]
> I suppose user might've reset the request (CMD+I) which will reset the request name to `Untitled request`.

Nope, I checked it again and am able to reproduce the issue. This name `Untitled request` comes from `makeRESTRequest` in `curlparser.ts` from the name key.

**Comment by kyteinsky** [comment]
https://user-images.githubusercontent.com/20724224/155871257-8a32d5c6-07b2-4b67-b20e-fec41c94eb8d.mp4

**Comment by AndrewBastin** [comment]
Oh okay.

Now, to be honest, I have found many issues with the Curl Implementation after the PR was merged. We might have to work on a continuing PR to rethink and reorganize some bits. This time with stronger standards and testing implementation to reinforce things are working properly.

**Comment by kyteinsky** [comment]
I did test the curl parser extensively, but this definitely calls for thorough tests to be written.

With that aside, I don't believe this issue has to do anything with that PR. `makeRESTRequest` and `setRESTRequest` are causing this to happen. I need some more time to look through it.
